### ISSUE TITLE
A potential Issue in Forward ECMC

### ISSUE THREAD

**Issue by 162348** [charlyandral/pdmp_jax#1]
First of all, thanks for your great work!

There is a normalization step when performing a velocity jump in the Forward Event Chain Monte Carlo implementation.

https://github.com/charlyandral/pdmp_jax/blob/c39820c129f883a5916e7d6e300dbf6457768ba3/src/pdmp_jax/samplers/ForwardEventChain.py#L89

But if the algorithm is correctly implemented, I believe the following code will produce a reducible Markov process on $\mathbb{R}^3$, being confined to one dimensional subspace $\mathbb{R}\cdot\mathtt{v}_{\mathtt{init}}$ instead of exploring the entire space.

This doesn't seem to be the case with your current implementation. I wonder why this implementation successfully explore the entire space, although the original paper suggests otherwise.

```python
import jax.numpy as jnp
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import pdmp_jax as pdmp

def U(x):
    return jnp.sum(x**2)/2

dim, grid_size, N_sk, N = 3, 10, 1000000, 1000000
grad_U = jax.grad(U)
xinit, vinit = jnp.zeros((dim,)), jnp.ones((dim,))
sampler = pdmp.ForwardEventChain(dim, grad_U, grid_size)
out = sampler.sample_skeleton(N_sk, xinit, vinit, verbose = True, seed=123)
sample = sampler.sample_from_skeleton(N,out)
plt.figure()
sns.jointplot(x = sample[:,0],y = sample[:,1])
plt.show()

pdmp.plot(out)
```

Here, $U$ is the negative log-likelihood of an isotropic Gaussian distribution, and for all $x\in\mathbb{R}^3$, $v_{\mathtt{init}}\parallel\nabla U(x)$ holds. Since the refreshment strategy of FECMC will just replace two (random) $\nabla U(x)$-tangential components, $v$'s $\nabla U(x)$-tangential component will remain zero if if started zero. Therefore, $v_t$ will always remain parallel to $v_{\texttt{init}}$, if my understanding is correct.

**Comment by 162348** [comment]
I believe I've identified the cause of the issue, and it turns out to be actually a noteworthy bug.

## Inspection

Below is an excerpt of the output from the (above) code:

```zsh
>>> out.x[0:4,:]
Array([[ 0.        ,  0.        ,  0.        ],
       [ 0.34932306,  0.34932306,  0.34932306],
       [-1.0266808 , -1.0266814 , -1.0266817 ],
       [-0.27029285, -0.05068687,  0.9752631 ]], dtype=float32)
>>> out.v[0:4,:]
Array([[ 1.        ,  1.        ,  1.        ],
       [-0.18454847, -0.18454856, -0.18454859],
       [ 0.29520535,  0.38091406,  0.781325  ],
       [ 0.65371704,  0.42848644, -0.62374127]], dtype=float32)
```

Notice that in the first two rows (`out.x[0:1,:]` and `out.v[0:1,:]`), all three columns share identical values, whereas starting from `out.v[2,:]` the columns differ significantly.

[This animation](https://github.com/162348/PDMPFlux.jl/blob/main/examples/SlantedGauss/ForwardECMC_StandardGauss3D.gif) might nicely illustrate the sampler's behaviour.

## Cause

Summary: This phenomenon is caused by the normalization step, amplifying the numerical errors -- by a factor of $10^{16}$ -- causing a sudden and erroneous change from `out.v[1,:]` to `out.v[2,:]`.

When $v$ is closely aligned with $\nabla U(x)$, the following computation can result in a really small but non-zero value:

https://github.com/charlyandral/pdmp_jax/blob/c39820c129f883a5916e7d6e300dbf6457768ba3/src/pdmp_jax/samplers/ForwardEventChain.py#L91

Later, `v_ortho` is orthogonally transformed into `v_prop`. Since the orthogonal transformation doesn't change the length of the vectors, `v_prop` also ends up with a very small yet non-zero norm. Then there comes this line of code:

https://github.com/charlyandral/pdmp_jax/blob/c39820c129f883a5916e7d6e300dbf6457768ba3/src/pdmp_jax/samplers/ForwardEventChain.py#L107

Finally, `v_prop` makes a non-trivial contribution that imparts a noticeable $\nabla U(x)$-perpendicular component to `v_out`, which should not occur in Forward ECMCs.

https://github.com/charlyandral/pdmp_jax/blob/c39820c129f883a5916e7d6e300dbf6457768ba3/src/pdmp_jax/samplers/ForwardEventChain.py#L117
